### PR TITLE
fix(ssh-keys): integrate with project as a boundary and integ test boundary

### PR DESCRIPTION
**Pull request by maghirardelli**
Issue #, if available:
GALI-1935/2290

Description of changes:
Integrate SSH Key Service with project as a boundary as Dynamic Authz middleware
Integration tests for whole scope of dynamic authz middleware to test the boundary

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts [hunk 1]
```diff
@@ -4,111 +4,154 @@
...
+    test.each(testBundle)('within a valid project', async (testCase) => {
+      const { username, session: sessionFunc, projectId: projectIdFunc } = testCase;
+      const session = sessionFunc();
+      const projectId = projectIdFunc();
+
+      console.log(`as ${username}`);
+
       try {
-        await adminSession.resources.projects.project(project.id).sshKeys().sshKey(existingSshKeyId).purge();
+        await session.resources.projects.project(projectId).sshKeys().sshKey(invalidSshKeyId).delete();
```
Comment (FernandoAranda) [inline solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:56]: I think this should be a purge

**Review by SanketD92 — APPROVED** [review]

File: solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts (hunk 1 above)
Comment (FernandoAranda) [reply solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:56]: I think Kevin just merged that code, thats why in the conflict the purge was there

Comment (janeyuaws) [reply solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts:56]: looks like the latest have both delete and purge? Looks like the one corresponding to sshKey delete is delete.

Comment (janeyuaws) [reply solutions/swb-reference/integration-tests/tests/isolated/sshKeys/delete.test.ts]: switched to purge

**Review by FernandoAranda — APPROVED** [review]

**Pull request merged**